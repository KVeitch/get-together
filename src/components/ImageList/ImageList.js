import React from 'react';
import PropTypes from 'prop-types';

export const ImageList = ({ images }) => {
  const imageList = images.map((url, i) => (
    <img src={url} alt="sites to see together" key={i} />
  ));

  return (
    <section className="section__images">
      <h3>Sites to see together!</h3>
      {imageList}
    </section>
  );
};

export default ImageList;

ImageList.propTypes = {
  images: PropTypes.array
};
