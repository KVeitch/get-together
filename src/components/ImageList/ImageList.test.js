import React from 'react';
import { shallow } from 'enzyme';
import ImageList from './ImageList';

describe('ImageList ', () => {
  const mockImages = [
    'https://images.unsplash.com/photo-1481315555141-253bc71d6f67',
    'https://images.unsplash.com/photo-1566928661283-35dc6abad9ef',
    'https://images.unsplash.com/photo-1532556884697-26324ee63594',
    'https://images.unsplash.com/photo-1545855869-f42702d73d91',
    'https://images.unsplash.com/photo-1545855869-fdd7d4a9753c',
    'https://images.unsplash.com/photo-1545855869-e3bbd54ff4bb',
    'https://images.unsplash.com/photo-1479150328630-b0307b442238',
    'https://images.unsplash.com/photo-1512454049206-bf6d9e5ccc97',
    'https://images.unsplash.com/uploads/14113833185419659518b/db0d71dd',
    'https://images.unsplash.com/photo-1437322810648-079dd4730a3b',
    'https://images.unsplash.com/photo-1536869186265-21d6e115c51a',
    'https://images.unsplash.com/photo-1545238554-660f7944fff4',
    'https://images.unsplash.com/photo-1518138945951-647273bf05aa',
    'https://images.unsplash.com/photo-1534239097568-685ece0e15bf',
    'https://images.unsplash.com/photo-1531166416685-ca5aa511eca1'
  ];
  const wrapper = shallow(<ImageList images={mockImages} />);
  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
