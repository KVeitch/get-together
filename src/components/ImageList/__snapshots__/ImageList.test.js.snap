// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList  should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="section__images"
>
  <h3>
    Sites to see together!
  </h3>
  <img
    alt="sites to see together"
    key="0"
    src="https://images.unsplash.com/photo-1481315555141-253bc71d6f67"
  />
  <img
    alt="sites to see together"
    key="1"
    src="https://images.unsplash.com/photo-1566928661283-35dc6abad9ef"
  />
  <img
    alt="sites to see together"
    key="2"
    src="https://images.unsplash.com/photo-1532556884697-26324ee63594"
  />
  <img
    alt="sites to see together"
    key="3"
    src="https://images.unsplash.com/photo-1545855869-f42702d73d91"
  />
  <img
    alt="sites to see together"
    key="4"
    src="https://images.unsplash.com/photo-1545855869-fdd7d4a9753c"
  />
  <img
    alt="sites to see together"
    key="5"
    src="https://images.unsplash.com/photo-1545855869-e3bbd54ff4bb"
  />
  <img
    alt="sites to see together"
    key="6"
    src="https://images.unsplash.com/photo-1479150328630-b0307b442238"
  />
  <img
    alt="sites to see together"
    key="7"
    src="https://images.unsplash.com/photo-1512454049206-bf6d9e5ccc97"
  />
  <img
    alt="sites to see together"
    key="8"
    src="https://images.unsplash.com/uploads/14113833185419659518b/db0d71dd"
  />
  <img
    alt="sites to see together"
    key="9"
    src="https://images.unsplash.com/photo-1437322810648-079dd4730a3b"
  />
  <img
    alt="sites to see together"
    key="10"
    src="https://images.unsplash.com/photo-1536869186265-21d6e115c51a"
  />
  <img
    alt="sites to see together"
    key="11"
    src="https://images.unsplash.com/photo-1545238554-660f7944fff4"
  />
  <img
    alt="sites to see together"
    key="12"
    src="https://images.unsplash.com/photo-1518138945951-647273bf05aa"
  />
  <img
    alt="sites to see together"
    key="13"
    src="https://images.unsplash.com/photo-1534239097568-685ece0e15bf"
  />
  <img
    alt="sites to see together"
    key="14"
    src="https://images.unsplash.com/photo-1531166416685-ca5aa511eca1"
  />
</section>
`;
